generic-sensor/generic-sensor-tests.js

```diff
@@ -36,84 +36,81 @@ function reading_to_array(sensor) {
+  promise_test(async t => {
+    const sensor1 = new sensorType();
+    const sensor2 = new sensorType();
+    const eventWatcher = new EventWatcher(t, sensor1, ["reading", "error"]);
     sensor2.start();
     sensor1.start();
+
+    await eventWatcher.wait_for("reading");
+    // Reading values are correct for both sensors.
+    assert_reading_not_null(sensor1);
+    assert_reading_not_null(sensor2);
+
+    //After first sensor stops its reading values are null,
+    //reading values for the second sensor remains
+    sensor1.stop();
+    assert_reading_null(sensor1);
+    assert_reading_not_null(sensor2);
+    sensor2.stop();
+    assert_reading_null(sensor2);
   }, `${sensorType.name}: sensor reading is correct`);
 
-  async_test(t => {
-    let sensor = new sensorType();
-    let cachedTimeStamp1;
-    sensor.onreading = () => {
-      cachedTimeStamp1 = sensor.timestamp;
-    };
-    sensor.onerror = t.step_func_done(unreached);
+  promise_test(async t => {
+    const sensor = new sensorType();
+    const eventWatcher = new EventWatcher(t, sensor, ["reading", "error"]);
     sensor.start();
-    t.step_timeout(() => {
-      sensor.onreading = t.step_func_done(() => {
-        //sensor.timestamp changes.
-        let cachedTimeStamp2 = sensor.timestamp;
-        assert_greater_than(cachedTimeStamp2, cachedTimeStamp1);
-        sensor.stop();
-      });
-    }, 1000);
+
+    await eventWatcher.wait_for("reading");
+    const cachedTimeStamp1 = sensor.timestamp;
+
+    await eventWatcher.wait_for("reading");
+    const cachedTimeStamp2 = sensor.timestamp;
+
+    assert_greater_than(cachedTimeStamp2, cachedTimeStamp1);
+    sensor.stop();
   }, `${sensorType.name}: sensor timestamp is updated when time passes`);
 
-  async_test(t => {
-    let sensor = new sensorType();
-    sensor.onerror = t.step_func_done(unreached);
+  promise_test(async t => {
+    const sensor = new sensorType();
+    const eventWatcher = new EventWatcher(t, sensor, ["reading", "error"]);
     assert_false(sensor.activated);
-    sensor.onreading = t.step_func_done(() => {
-      assert_true(sensor.activated);
-      sensor.stop();
-      assert_false(sensor.activated);
-    });
     sensor.start();
     assert_false(sensor.activated);
+
+    await eventWatcher.wait_for("reading");
+    assert_true(sensor.activated);
+
+    sensor.stop();
+    assert_false(sensor.activated);
   }, `${sensorType.name}: Test that sensor can be successfully created and its states are correct.`);
 
   test(() => {
-    let sensor, start_return;
-    sensor = new sensorType();
+    const sensor = new sensorType();
     sensor.onerror = unreached;
-    start_return = sensor.start();
+    const start_return = sensor.start();
     assert_equals(start_return, undefined);
     sensor.stop();
   }, `${sensorType.name}: sensor.start() returns undefined`);
 
   test(() => {
     try {
-      let sensor = new sensorType();
+      const sensor = new sensorType();
       sensor.onerror = unreached;
```

Comment:
`onerror` is async, should use `async_test` instead of `test`

Reply:
Use promise_test instead of test now.